File: app/src/main/java/com/lubenard/oring_reminder/utils/UiUtils.java
package com.lubenard.oring_reminder.utils;

import android.app.DatePickerDialog;
import android.app.TimePickerDialog;
import android.content.Context;
import android.text.format.DateFormat;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.lubenard.oring_reminder.R;

import java.util.Calendar;

public class UiUtils {
    /**
     * Show toast if a date has been malformed
     */
    public static void showToastBadFormattedDate(Context context) {
        Toast.makeText(context, R.string.bad_date_format, Toast.LENGTH_SHORT).show();
    }

    public static void enableEditText(EditText editText) {
        editText.setFocusable(true);
        editText.setFocusableInTouchMode(true);
    }

    public static void disableEditText(EditText editText) {
        editText.setFocusable(false);
    }


    /**
     * Open time picker
     * @param filling_textview
     */
    public static void openTimePicker(Context context, TextView filling_textview, boolean showTextViewDate) {
        // Get Current Time
        final Calendar c = Calendar.getInstance();
        int mHour;
        int mMinute;
        if (showTextViewDate) {
            String[] timePutSplitted = filling_textview.getText().toString().split(":");
            mHour = Integer.parseInt(timePutSplitted[0]);
            mMinute = Integer.parseInt(timePutSplitted[1]);
        } else {
            mHour = c.get(Calendar.HOUR_OF_DAY);
            mMinute = c.get(Calendar.MINUTE);
        }
        // Launch Time Picker Dialog
        TimePickerDialog timePickerDialog = new TimePickerDialog(context, (view, hourOfDay, minute) -> filling_textview.setText(hourOfDay + ":" + minute + ":00"), mHour, mMinute, DateFormat.is24HourFormat(context));
        timePickerDialog.show();
    }

    /**
     * Open Calendar picker
     * @param filling_textview
     */
    public static void openCalendarPicker(Context context, TextView filling_textview, boolean showTextViewDate) {
        // Get Current Date
        final Calendar c = Calendar.getInstance();
        int mYear;
        int mMonth;
        int mDay;
        if (showTextViewDate) {
            String[] timePutSplitted = filling_textview.getText().toString().split("-");
            mYear = Integer.parseInt(timePutSplitted[0]);
            mMonth = Integer.parseInt(timePutSplitted[1]);
            mDay = Integer.parseInt(timePutSplitted[2]);
        } else {
            mYear = c.get(Calendar.YEAR);
            mMonth = c.get(Calendar.MONTH);
            mDay = c.get(Calendar.DAY_OF_MONTH);
        }

        DatePickerDialog datePickerDialog = new DatePickerDialog(context,
                (view, year, monthOfYear, dayOfMonth) -> filling_textview.setText(year + "-" + (monthOfYear + 1) + "-" + dayOfMonth), mYear, mMonth, mDay);
        datePickerDialog.show();
    }
}

File: app/src/main/java/com/lubenard/oring_reminder/utils/UiUtils.kt
package com.lubenard.oring_reminder.utils

import android.app.DatePickerDialog
import android.app.TimePickerDialog
import android.content.Context
import android.text.format.DateFormat
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

import com.lubenard.oring_reminder.R

import java.util.Calendar

class UiUtils {
    companion object {

        /**
         * Show toast if a date has been malformed
         */
        fun showToastBadFormattedDate(context: Context) {
            Toast.makeText(context, R.string.bad_date_format, Toast.LENGTH_SHORT).show()
        }

        fun enableEditText(editText: EditText) {
            editText.isFocusable = true
            editText.isFocusableInTouchMode = true
        }

        fun disableEditText(editText: EditText) {
            editText.isFocusable = false
        }

        /**
         * Open time picker
         * @param filling_textview
         */
        fun openTimePicker(
            context: Context,
            filling_textview: TextView,
            showTextViewDate: Boolean
        ) {
            // Get Current Time
            val calendar = Calendar.getInstance()
            val mHour: Int
            val mMinute: Int
            val timePutSplitted = filling_textview.text.toString().split(":")

            if (showTextViewDate && timePutSplitted.size == 2) {
                mHour = Integer.parseInt(timePutSplitted[0])
                mMinute = Integer.parseInt(timePutSplitted[1])
            } else {
                mHour = calendar.get(Calendar.HOUR_OF_DAY)
                mMinute = calendar.get(Calendar.MINUTE)
            }
            // Launch Time Picker Dialog
            val timePickerDialog = TimePickerDialog(
                context,
                { _, hourOfDay, minute ->
                    filling_textview.text = "$hourOfDay:$minute:00"
                },
                mHour,
                mMinute,
                DateFormat.is24HourFormat(context)
            )
            timePickerDialog.show()
        }

        /**
         * Open Calendar picker
         * @param filling_textview
         */
        fun openCalendarPicker(
            context: Context,
            filling_textview: TextView,
            showTextViewDate: Boolean
        ) {
            // Get Current Date
            val calendar = Calendar.getInstance()
            val mYear: Int
            val mMonth: Int
            val mDay: Int
            val timePutSplitted = filling_textview.text.toString().split("-")
            if (showTextViewDate && timePutSplitted.size == 3) {
                mYear = Integer.parseInt(timePutSplitted[0])
                mMonth = Integer.parseInt(timePutSplitted[1])
                mDay = Integer.parseInt(timePutSplitted[2])
            } else {
                mYear = calendar.get(Calendar.YEAR)
                mMonth = calendar.get(Calendar.MONTH)
                mDay = calendar.get(Calendar.DAY_OF_MONTH)
            }

            val datePickerDialog = DatePickerDialog(
                context,
                { _, year, monthOfYear, dayOfMonth ->
                    filling_textview.text =
                        "$year-${if (monthOfYear < 10) "0" else ""}${(monthOfYear + 1)}-$dayOfMonth"
                },
                mYear, mMonth - 1, mDay
            )
            datePickerDialog.show()
        }
    }
}
